### PR TITLE
RELATED: SD-1050 Add new ad event type visualizationCancelled to sdk-embedding

### PR DESCRIPTION
**Pull request by lanhq**
JIRA: SD-1050

---

Supported PR commands:

| Command         | Description            |
| --------------- | ---------------------- |
| `ok to test`    | Re-run standard checks |
| `extended test` | BackstopJS tests       |

---

# PR Checklist

-   [x] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [x] `check` passes
-   [x] `check-extended` passes
-   [x] (not applicable) `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by lanhq** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/333/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/1924/ : SUCCESS in 12m 49s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/334/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/248/ : SUCCESS in 16m 22s

File: libs/sdk-embedding/src/iframe/ad.ts [hunk 1]
```diff
@@ -176,6 +176,11 @@ export namespace EmbeddedAnalyticalDesigner {
          * Type represent that the remove filter context action is finished
          */
         RemoveFilterContextFinished = "removeFilterContextFinished",
+
+        /**
+         * Type notify KD that the editing insight has been canceled
+         */
+        VisualizationCancelled = "visualizationCancelled",
```
Comment (no23reason) [inline libs/sdk-embedding/src/iframe/ad.ts:183]: The name is a bit confusing to me, it does not mention anything about editting. Should it be something like "insightEdittingCanceled"?

File: libs/sdk-embedding/src/iframe/ad.ts [hunk 2]
```diff
@@ -176,6 +176,11 @@ export namespace EmbeddedAnalyticalDesigner {
          * Type represent that the remove filter context action is finished
          */
         RemoveFilterContextFinished = "removeFilterContextFinished",
+
+        /**
+         * Type notify KD that the editing insight has been canceled
```
Comment (no23reason) [inline libs/sdk-embedding/src/iframe/ad.ts:181]: I do not think we need to mention KD here, others can use this event as well, right?

Comment (lanhq) [reply libs/sdk-embedding/src/iframe/ad.ts:181]: For the feature https://jira.intgdc.com/browse/GDP-871, KD will open AD overlay. Then the event `VisualizationCancelled` will be sent to KD to close the AD overlay. And It is used internally.

File: libs/sdk-embedding/src/iframe/ad.ts (hunk 1 above)
Comment (lanhq) [reply libs/sdk-embedding/src/iframe/ad.ts:183]: The name `insightEdittingCanceled` look better than `visualizationCancelled`. What do you think @vu-pham ?

Comment (vu-pham) [reply libs/sdk-embedding/src/iframe/ad.ts:183]: I agree, and it should be insightEdittingCance**ll**ed

Comment (lanhq) [reply libs/sdk-embedding/src/iframe/ad.ts]: correct name insightEdit**T**ingCanceled -> `insightEditingCanceled`

File: libs/sdk-embedding/src/iframe/ad.ts (hunk 2 above)
Comment (no23reason) [reply libs/sdk-embedding/src/iframe/ad.ts]: Ok, thanks for the explanation :)

File: libs/sdk-embedding/src/iframe/ad.ts [hunk 3]
```diff
@@ -176,6 +176,11 @@ export namespace EmbeddedAnalyticalDesigner {
          * Type represent that the remove filter context action is finished
          */
         RemoveFilterContextFinished = "removeFilterContextFinished",
+
+        /**
+         * Type notify KD that the insight editing has been canceled
+         */
+        InsightEditingCanceled = "insightEditingCanceled",
```
Comment (no23reason) [inline libs/sdk-embedding/src/iframe/ad.ts:183]: Typo: cancelled (double L)

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/239/ : SUCCESS in 20s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/494/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/1929/ : SUCCESS in 19m 33s

**Pull request merged**